zulip/zulip/__init__.py

```diff
@@ -526,9 +523,6 @@ def error_retry(error_string):
                     sys.stdout.write(".")
                 sys.stdout.flush()
             query_state["request"]["dont_block"] = json.dumps(True)
-            time.sleep(1)
-            query_state["failures"] += 1
-            return True
```

Comment:
Why not just have this function do the `backoff.file()` and `return backoff.keep_going()` ?  It's deduplicate code.

Reply:
This is a really nice idea. Implemented this and one other change and now my do_api_query commit is way more readable and has minimal changes.